### PR TITLE
Cleanup dart analyze diagnostics in ignored directories.

### PR DESCRIPTION
**Pull request by matanlurey**
Towards https://github.com/flutter/flutter/issues/152636.

I also deleted some code that was TODO'd 10 years ago.

### CODE REVIEW

File: shell/testing/observatory/service_client.dart [hunk 1]
```diff
@@ -44,55 +47,46 @@ class ServiceClient {
...
     final dynamic key = response['id'];
     if (key != null) {
       print('<- $key');
-      final dynamic completer = _outstandingRequests.remove(key);
-      assert(completer != null);
-      final dynamic result = response['result'];
-      final dynamic error = response['error'];
+      final completer = _outstandingRequests.remove(key)!;
+      final result = response['result'];
+      final error = response['error'];
```
Comment (gaaclarke) [inline shell/testing/observatory/service_client.dart:56]: Should dynamic be explicit?  I guess we don't have a lint, but it is explicit above.

File: shell/testing/observatory/test.dart [hunk 2]
```diff
@@ -57,114 +31,53 @@ Future<String> readResponse(HttpClientResponse response) {
...
   final HttpClient client = HttpClient();
   final HttpClientRequest request = await client.getUrl(uri);
   final HttpClientResponse response = await request.close();
   Expect.equals(response.statusCode, 200);
-  await response.drain();
+  await response.drain<void>();
   client.close();
 }
 
-Future<Null> testStartPaused(Uri uri) async {
```
Comment (gaaclarke) [inline shell/testing/observatory/test.dart:99]: Why is this gone now?

Comment (matanlurey) [reply shell/testing/observatory/test.dart:99]: Dead code that's been disabled for about 6 or 7 years

File: shell/testing/observatory/service_client.dart (hunk 1 above)
Comment (gaaclarke) [reply shell/testing/observatory/service_client.dart:56]: It's `Object?`? Sounds good.

**Review by gaaclarke — APPROVED** [review]
lgtm

**Pull request merged**